File: java/src/net/razorvine/pickle/InvalidOpcodeException.java
package net.razorvine.pickle;

/**
 * Exception thrown when the unpickler encounters an invalid opcode.
 * 
 * @author Irmen de Jong (irmen@razorvine.net)
 */
public class InvalidOpcodeException extends PickleException {

	private static final long serialVersionUID = -7846953716411277176L;

	public InvalidOpcodeException(String message, Throwable cause) {
		super(message, cause);
	}

	public InvalidOpcodeException(String message) {
		super(message);
	}

}

File: dotnet/Pyrolite/Pickle/InvalidOpcodeException.cs
/* part of Pyrolite, by Irmen de Jong (irmen@razorvine.net) */

using System;
using System.Runtime.Serialization;

namespace Razorvine.Pickle
{
	/// <summary>
	/// Exception thrown when the unpickler encountered an unknown or unimplemented opcode.
	/// </summary>
	public class InvalidOpcodeException : PickleException
	{
		public InvalidOpcodeException()
		{
		}

	 	public InvalidOpcodeException(string message) : base(message)
		{
		}

		public InvalidOpcodeException(string message, Exception innerException) : base(message, innerException)
		{
		}

		// This constructor is needed for serialization.
		protected InvalidOpcodeException(SerializationInfo info, StreamingContext context) : base(info, context)
		{
		}
	}
}